components/sync15/src/error.rs

```diff
@@ -23,12 +21,33 @@ pub enum ErrorResponse {
 
 #[derive(Debug, thiserror::Error)]
 pub enum ErrorKind {
+    // These are errors duplicated from SyncTraitsError, so that consumers can
+    // deal with errors from just 1 of the crates and not care which one of
+    // then actually caused the error.
     #[error("Key {0} had wrong length, got {1}, expected {2}")]
     BadKeyLength(&'static str, usize, usize),
 
     #[error("SHA256 HMAC Mismatch error")]
     HmacMismatch,
 
+    #[error("Crypto/NSS error: {0}")]
+    CryptoError(#[from] rc_crypto::Error),
+
+    #[error("Base64 decode error: {0}")]
+    Base64Decode(#[from] base64::DecodeError),
+
+    #[error("JSON error: {0}")]
+    JsonError(#[from] serde_json::Error),
+
+    #[error("Bad cleartext UTF8: {0}")]
+    BadCleartextUtf8(#[from] std::string::FromUtf8Error),
+
+    #[error("HAWK error: {0}")]
+    HawkError(#[from] rc_crypto::hawk::Error),
+
+    //
+    // Errors specific to this module.
+    //
```

Comment:
how do we feel about having only one variant like:
```rs
#[error("{0}")]
Sync15TraitsError(#[from] SyncTraitsError)
```

And just get rid of the `From` implementation.

The downside is that consumers would need to go one level deeper to get the "Actual" error though so feel free to disregard this

It would also be a breaking change though, so we'd have to make sure none of the variants deleted are consumed anywhere external

Reply:
This would still leave 2 different ways to spell the same error. In this patch as it stands, the tabs component can say `TabsError::SyncAdapterError::HmacMismatch` and it will match the error whether it originated in sync15 or sync15-traits. 

(I also agree with Ben's broader point that the split is a bad thing, so this would ideally help us re-uniffy things in the future)